lib/queries/otu_filter_query.rb

```diff
@@ -0,0 +1,215 @@
+    end
+
+    def a
+      @a
+    end
+
+    def initialize(params)
+      params.reject! { |k, v| v.blank? }
+
+      @query_params                 = params
+      @query_geographic_area_ids    = params[:geographic_area_ids]
+      @query_shape                  = params[:drawn_area_shape]
+      @query_author_ids             = params[:author_ids]
+      @query_verbatim_author_string = params[:verbatim_author_string]
+      @query_and_or_select          = params[:and_or_select]
+      @query_nomen_id               = params[:nomen_id]
+      @query_descendants            = params[:descendants]
+    end
+
+    def area_set?
+      !query_geographic_area_ids.nil?
+    end
+
+    def author_set?
+      retval = case query_author_ids
+                 when nil
+                   false
+                 else
+                   query_author_ids.count > 0
+               end
+      retval
+    end
+
+    def nomen_set?
+      !query_nomen_id.nil?
+    end
+
+    def verbatim_set?
+      !query_verbatim_author_string.blank?
+    end
+
+    def shape_set?
+      !query_shape.nil?
+    end
+
+    def with_descendants?
+      query_descendants == '_on_'
+    end
+
+    # All scopes might end up in Otu directly
+
+    # @return [Scope]
+    # def otu_scope
+    #   # Challenge: Refactor to use a join pattern instead of SELECT IN
+    #   innerscope = with_descendants? ? Otu.self_and_descendants_of(query_nomen_id) : Otu.where(id: query_nomen_id)
+    #   # Otu.where(id: innerscope)
+    #   innerscope
+    # end
+
+=begin
+      1. find all geographic_items in area(s)/shape
+      2. find all georeferences which are associated with result #1
+      3. find all collecting_events which are associated with result #2
+      4. find all collection_objects which are associated with result #3
+      5. find all otus which are associated with result #4
+=end
+    # @return [Scope]
+    def geographic_area_scope
+      # This could be simplified if the AJAX selector returned a geographic_item_id rather than a GeographicAreaId
+      target_geographic_item_ids = []
+
+      query_geographic_area_ids.each do |gaid|
+        target_geographic_item_ids.push(
+          GeographicArea.
+            joins(:geographic_items).
+            find(gaid).
+            default_geographic_item.id
+        )
+      end
+      # r4 = CollectionObject.joins(:geographic_items)
+      #        .where(GeographicItem.contained_by_where_sql(target_geographic_item_ids))
+      r42i = CollectionObject.joins(:geographic_items)
+               .where(GeographicItem.contained_by_where_sql(target_geographic_item_ids))
+               .distinct
+               .pluck(:id) # < - bad
+      # get the Otus associated with r4
+      # r5i = r4.map(&:otus).flatten.pluck(:id).uniq
+      # r5o = Otu.where(id: r5i)
+      r5o2 = Otu.joins(:collection_objects).where('biological_collection_object_id in (?)', r42i)
+      # r5o
+      r5o2
+    end
+
+    # @return [Scope]
+    def shape_scope
```

Comment:
cleanup comments, maybe slightly less crytpic variable name now, just provide scope at end of method for return